### PR TITLE
fix: propagate nested fields description

### PR DESCRIPTION
**Pull request by tiboun**
## Summary
Attach field comment to schema for element not on top level.

**PR Type: Bug Fix**

**Status: Ready to review**

**Breaking change? No**

### Solution
Attach description set from yml schema to spark schema.

### How has this been tested?
Locally with a target on a bq table containing nested fields and arrays.

### Other changes
- Add a provided dependency in order to ease debugging from IDE such as intellij.
- Remove unecessary field from Metadata class

## Contributor checklist:
Go over all the following points, and put an `x` in all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [X] My code follows the code style of this project.
- [X] I have updated the Release notes.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -121,6 +121,7 @@ object Dependencies {
     // see https://github.com/GoogleCloudDataproc/spark-bigquery-connector/issues/36
     // Add the jar file to spark dependencies
     "com.google.cloud.spark" %% "spark-bigquery-with-dependencies" % Versions.sparkBigqueryWithDependencies % "provided" excludeAll (jacksonExclusions: _*),
+    "com.google.cloud.spark" % "spark-bigquery-connector-common" % Versions.sparkBigqueryConnectorCommon % "provided" excludeAll (jacksonExclusions: _*),
```
Comment (hayssams) [inline project/Dependencies.scala:124]: What is this about ?

**Review by hayssams — APPROVED** [review]
LGTM

File: project/Dependencies.scala (hunk 1 above)
Comment (tiboun) [reply project/Dependencies.scala:124]: This is a dependency used by spark bigquery. Without this, I don't get source code automatically while debugging.

Comment (hayssams) [reply project/Dependencies.scala:124]: Good to know

**Pull request merged**